tensorflow/workspace2.bzl

```diff
@@ -114,6 +114,10 @@ def _tf_toolchains():
     # TODO: Remove def file filter when TensorFlow can export symbols properly on Windows.
     def_file_filter_configure(name = "local_config_def_file_filter")
 
+    # Register the toolchain to support clang-cl on Windows
+    native.register_toolchains("@local_config_cc//:cc-toolchain-x64_windows-clang-cl")
```

Comment:
This needs to be applied only on Windows and only if clang is enabled. Our other platforms are getting presubmit errors like:

```
ERROR: /[tf/tensorflow/tensorflow/python/compiler/xla/BUILD:26](https://cs.corp.google.com/piper///depot/google3/tf/tensorflow/tensorflow/python/compiler/xla/BUILD?l=26&cl=565719930):20: While resolving toolchains for target 
//tensorflow/python/compiler/xla:jit_test_cpu: invalid registered toolchain '@local_config_cc//:cc-toolchain-x64_windows-clang-cl': no such target '@local_config_cc//:cc-toolchain-x64_windows-clang-cl': target 'cc-toolchain-x64_windows-clang-cl' not declared in package '' defined by 
/root/.cache/bazel/_bazel_root/fbac33eb30dbfb6b11b15a7ff5ac830d/external/local_config_cc/BUILD (Tip: use `query "@local_config_cc//:*"` to see all the targets in that package)
```

Reply:
@angerson, I have made changes to fix the pre-submit error. Please review and validate, and let me know if it still fails or if there is any other failure